check_tags_in_requests.py

```diff
@@ -106,8 +109,34 @@ def checkTagInRequest(self, req, a):
 
         return True
 
+    def isNewPackage(self, tgt_project, tgt_package):
+        try:
+            osc.core.show_devel_project(self.apiurl, tgt_project, tgt_package)
+        except HTTPError:
+            return True
+        return False
+
+    def checkTagNotRequired(self, req, a):
+        # A tag is not required only if the package is
+        # already in Factory with the same revision,
+        # and the package is being introduced, not updated
+        factory_checker = check_source_in_factory.FactorySourceChecker(apiurl=self.apiurl,
+                                                                       dryrun=self.dryrun,
+                                                                       logger=self.logger,
+                                                                       user=self.review_user,
+                                                                       group=self.review_group)
+        factory_ok = factory_checker.check_source_submission(a.src_project, a.src_package, a.src_rev,
```

Comment:
you  may want to check that first and return right away to avoid the potentially expensive factory source check.